Add 'Context Object' page (#525)
comcode-org/hackmud_wiki: docs/scripting/syntax/context.mdx

---
title: Context Object
---

> The first argument to the script function is known as the "context object" or "context".

Reviewer: ```suggestion
The first argument given to script functions is known as the "context object" or "context".
```

## Syntax

The context object always contains the following keys:

> - `caller`: the name of the user that initiated this script execution, as string.

Reviewer: ```suggestion
- `caller`: the name of the user that initiated this script execution, as a string.
```

- `calling_script`: the full name of the script that called this script, as a string. This is only filled if the script was called via [[subscript]]. If the script was called from the CLI or via scriptor, this value is (( %Vnull% )) instead.
- `this_script`: the full name of this script, as a string.
- `cols`: the width of the main shell in the client running this script in characters, as a number. If called by automation (such as a [[cron_bot]] execution), (( %V128% )) is used.
- `rows`: the height of the main shell in the client running this script in characters, as a number. If called by automation, (( %V64% )) is used.

The following properties may also be present:

- `is_scriptor`: (( %Vtrue% )) if the script is being run via a scriptor.
- `is_brain`: (( %Vtrue% )) if the script execution was started by a [[cron_bot]].

## Examples

```js
{
  caller: "myuser",
  calling_script: null,
  this_script: "myuser.test_script",
  cols: 125,
  rows: 78
}
```

```js
{

>   caller: "mycron",

Reviewer: ```suggestion
  caller: "myuser",
``` 
it might make sense to keep this as `myuser` instead of `mycron` to prevent confusion in the examples list?

Author: I intentionally switched users here to be a bit more "this is different", but now I'm not sure if that's necessary. I'm open to discussion on this.

Reviewer 2: We talked about this in office hours. I don't think it's a bad idea to call out a separate user here (with different username), but cron- in particular, requires more understanding than something like 'bot'

I also think that we should lean in a bit more on the hackmud 'energy' and name these users something like `my_cool_user` and `my_cool_bot_user`

  calling_script: null,

>   this_script: "mycron.bot_brain",

Reviewer: ```suggestion
  this_script: "myuser.bot_brain",
``` 
ditto
